ansible_collections/f5networks/f5_modules/plugins/modules/bigip_virtual_server.py

```diff
@@ -1991,7 +1991,10 @@ def profiles(self):
                 if 'name' not in profile:
                     tmp['name'] = profile
                 if 'partition' not in profile:
-                    tmp['partition'] = "Common"
+                    if len(profile["name"].split("/")) > 1:
+                        tmp["partition"] = profile["name"].split("/")[1]
+                    else:
+                        tmp['partition'] = "Common"
```

Comment:
Unfortunately due to some bad design choices profile parameter is of type 'raw' not list with dict type elements:

https://github.com/F5Networks/f5-ansible/blob/630e066016d044e5c971115ac9a6637be89ce0bb/ansible_collections/f5networks/f5_modules/plugins/modules/bigip_virtual_server.py#L3643

This means that Ansible does not enforce presence of name/context keys. So users can do things like here:
https://github.com/F5Networks/f5-ansible/blob/630e066016d044e5c971115ac9a6637be89ce0bb/ansible_collections/f5networks/f5_modules/plugins/modules/bigip_virtual_server.py#L547

You need to check for existence of 'name' key in profile, otherwise if you will get key error. Make the change to cover this scenario too.